lib/bson/parser/serializer.js

```diff
@@ -371,7 +371,8 @@ var serializeDecimal128 = function(buffer, key, value, index, isArray) {
   index = index + numberOfWrittenBytes;
   buffer[index++] = 0;
   // Write the data from the value
-  value.bytes.copy(buffer, index, 0, 16);
+  if (value.bytes.copy) value.bytes.copy(buffer, index, 0, 16);
```

Comment:
This might be premature optimization, but can't we cache this check far earlier? E.g. at the top like `var hasCopy = Buffer.prototype.copy != null`